libnetdata/required_dummies.h

```diff
@@ -0,0 +1,38 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+#ifndef NETDATA_LIB_DUMMIES_H
+#define NETDATA_LIB_DUMMIES_H 1
+
+// callback required by fatal()
+void netdata_cleanup_and_exit(int ret)
+{
+    exit(ret);
+}
+
+void send_statistics(const char *action, const char *action_result, const char *action_data)
+{
+    (void)action;
+    (void)action_result;
+    (void)action_data;
+    return;
+}
+
+// callbacks required by popen()
+void signals_block(void){};
+void signals_unblock(void){};
+void signals_reset(void){};
+
+// callback required by eval()
+int health_variable_lookup(const char *variable, uint32_t hash, struct rrdcalc *rc, calculated_number *result)
+{
+    (void)variable;
+    (void)hash;
+    (void)rc;
+    (void)result;
+    return 0;
+};
+
+// required by get_system_cpus()
+char *netdata_configured_host_prefix = "";
+
+#endif // NETDATA_LIB_DUMMIES_H
```

Comment:
I deleted this entire file and the tests run just fine. It should be removed.

Reply:
I can't build tests without these functions defined:
```
make  libnetdata/tests/str2ld_testdriver libnetdata/storage_number/tests/storage_number_testdriver
make[3]: Entering directory '/home/vlad/netdata'
  CC       libnetdata/tests/test_str2ld.o
  CCLD     libnetdata/tests/str2ld_testdriver
/usr/bin/ld: libnetdata/eval/eval.o: in function `eval_variable':
/home/vlad/netdata/libnetdata/eval/eval.c:170: undefined reference to `health_variable_lookup'
/usr/bin/ld: libnetdata/libnetdata.o: in function `verify_netdata_host_prefix':
/home/vlad/netdata/libnetdata/libnetdata.c:1207: undefined reference to `netdata_configured_host_prefix'
/usr/bin/ld: /home/vlad/netdata/libnetdata/libnetdata.c:1231: undefined reference to `netdata_configured_host_prefix'
/usr/bin/ld: /home/vlad/netdata/libnetdata/libnetdata.c:1235: undefined reference to `netdata_configured_host_prefix'
/usr/bin/ld: /home/vlad/netdata/libnetdata/libnetdata.c:1239: undefined reference to `netdata_configured_host_prefix'
/usr/bin/ld: /home/vlad/netdata/libnetdata/libnetdata.c:1208: undefined reference to `netdata_configured_host_prefix'
/usr/bin/ld: libnetdata/libnetdata.o:/home/vlad/netdata/libnetdata/libnetdata.c:1245: more undefined references to `netdata_configured_host_prefix' follow
/usr/bin/ld: libnetdata/log/log.o: in function `fatal_int':
/home/vlad/netdata/libnetdata/log/log.c:826: undefined reference to `send_statistics'
/usr/bin/ld: /home/vlad/netdata/libnetdata/log/log.c:828: undefined reference to `netdata_cleanup_and_exit'
/usr/bin/ld: libnetdata/os.o: in function `get_system_cpus':
/home/vlad/netdata/libnetdata/os.c:36: undefined reference to `netdata_configured_host_prefix'
/usr/bin/ld: libnetdata/os.o: in function `get_system_pid_max':
/home/vlad/netdata/libnetdata/os.c:93: undefined reference to `netdata_configured_host_prefix'
collect2: error: ld returned 1 exit status
make[3]: *** [Makefile:2443: libnetdata/tests/str2ld_testdriver] Error 1
make[3]: Leaving directory '/home/vlad/netdata'
make[2]: *** [Makefile:3939: check-am] Error 2
make[2]: Leaving directory '/home/vlad/netdata'
make[1]: *** [Makefile:3478: check-recursive] Error 1
make[1]: Leaving directory '/home/vlad/netdata'
make: *** [Makefile:3942: check] Error 2
[vlad@arch-esxi netdata]$
```

Reply:
Found it. It's the `--disable-lto` flag.
```
sudo git clean -dxf
sudo ./netdata-installer.sh --disable-lto
sudo make check
```
->  undefined references when you try building tests.

Reply:
I guess then we can't get rid of the file after all.